compiler/rustc_resolve/src/errors.rs

```diff
@@ -326,6 +326,18 @@ pub(crate) struct ParamInTyOfConstParam {
     #[label]
     pub(crate) span: Span,
     pub(crate) name: Symbol,
+    #[subdiagnostic]
+    pub(crate) sub_is_type: Option<ParamInTyOfConstParamIsType>,
```

Comment:
I don't understand this terminology. Is this the kind of the param that appears in type of const param?

Reply:
I'm not completely sure what the naming scheme is tbh I just copied it. I think the `sub` is short for "subdiagnostic", the `is_type` is to determine what kind of generic parameter was found. I've renamed all this to be more clear since I found it rather confusing too and so did others I spoke to.

It's now `pub(crate) param_kind: Option<ParamKindInTyOfConstParam>`.

As far as I know none of these `ParamKindIn<blank>` can be merged into a single type as they all have different diagnostic messages with their `#[note]`/`#[help]`. It's definitely unfortunate because there's a lot of them :sweat_smile: